Makefile.d/install-flannel.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -eu -o pipefail
+
+function INFO() {
+	echo >&2 -e "\e[104m\e[97m[INFO]\e[49m\e[39m $@"
+}
+function WARNING() {
+	echo >&2 -e "\e[101m\e[97m[WARNING]\e[49m\e[39m $@"
+}
+
+function ERROR() {
+	echo >&2 -e "\e[101m\e[97m[ERROR]\e[49m\e[39m $@"
+}
+
+# See chart values, 0 indicates default for platform
+# https://github.com/flannel-io/flannel/blob/v0.26.1/chart/kube-flannel/values.yaml
+: "${U7S_PORT_FLANNEL:='0'}"
+: "${U7S_PORT_ETCD:='2379'}"
+
+INFO "Flannel port: ${U7S_PORT_FLANNEL}"
+INFO "ETCD port: ${U7S_PORT_ETCD}"
+
+# Check hard dependency commands
+for cmd in helm kubectl git; do
+	if ! command -v "${cmd}" >/dev/null 2>&1; then
+		ERROR "Command \"${cmd}\" is not installed"
+		exit 1
+	fi
+done
+
+# We need to customize the values.yaml to expose the backendPort and args
+flannel_root=$(mktemp -d -u -t flannel-XXXXXXX)
+git clone --quiet --depth 1 --branch v0.26.1 https://github.com/flannel-io/flannel $flannel_root
+cd $flannel_root/chart
+
+# Write a new values.yaml that exposes what we need
+cat <<EOF > ./new-values.yaml
```

Comment:
Can we just use `yq` to replace `flannel.backendPort`?

Reply:
In the helm chart, we need to re-create the values.yaml because the [majority is commented out](https://github.com/flannel-io/flannel/blob/master/chart/kube-flannel/values.yaml#L27). And that's the only file that has that variable exposed to go into the config map. If we tried to edit the single install file, it would be a much more hacky operation as we would need to not just parse yaml (the seconds and values aren't there) but json objects within it. I think the approach here with helm that you suggested is the best strategy for now. We can engage further with the maintainers there to see if 1. they can vendor the helm chart to have values exposed we need (and then we can just use it) or 2. vendor the single yaml install to include more of the defaults (that are currently missing).

Reply:
No need to parse JSON if you apply `yq` to `values.yaml`.

Or maybe just use a plain old `diff` to patch `values.yaml`?

Reply:
I think I prefer the strategy that we currently have because it gives absolute transparency into the values.yaml that we are applying. I think what I'd like to do after the work here is follow up on the flannel repository and ask the developers to expose the values, and then we can just use theirs without our customization here. So we can consider this (ideally) a temporary fix. Is that OK?